tests/nightly/test_all.sh

```diff
@@ -44,6 +44,7 @@ USE_CUDA=1
 USE_CUDA_PATH=/usr/local/cuda
 USE_CUDNN=1
 USE_DIST_KVSTORE=1
+USE_ALLREDUCE_DIST_KVSTORE=1
```

Comment:
I'm not sure if this script is being used for nightly test at all. @marcoabreu could you verify? 
Also - do we need to add a build stage with USE_ALLREDUCE_DIST_KVSTORE=1 to Jenkins?

Reply:
I'm quite certain that we are not using this file
@mbaijal could you please confirm? We should be deleting this file otherwise.

Sure, feel free to add it. Does it need to be a separate build or could we just extend the current one? Please consider doing that in the nightly Jenkinsfile at tests/nightly/JenkinsfileForBinary (or something like that)